File: ngx-admin/src/app/model/category/CategoryUpdateRequestDTO.ts
import {MainCategoryDTO} from "./MainCategoryDTO";
import {SubCategoryDTO} from "./SubCategoryDTO";
import {ParametersDTO} from "./ParametersDTO";
import {SelectiveParameterValueDTO} from "./SelectiveParameterValueDTO";

export interface CategoryUpdateRequestDTO {
  mainCategoriesToUpdate: MainCategoryDTO[];
  subCategoriesToUpdate: SubCategoryDTO[];
  parametersToUpdate: ParametersDTO[];
  selectiveParametersToUpdate: SelectiveParameterValueDTO[];
  mainCategoryIdsToDelete: number[];
  subCategoryIdsToDelete: number[];
  parameterIdsToDelete: number[];
  selectiveParameterIdsToDelete: number[];
}

File: mingachevir-real-estate-server-master/src/main/java/io/mingachevir/mingachevirrealestateserver/model/dto/admin/CategoryUpdateRequestDTO.java
package io.mingachevir.mingachevirrealestateserver.model.dto.admin;

import lombok.Data;

import java.util.List;

@Data
public class CategoryUpdateRequestDTO {
    private List<MainCategoryDTO> mainCategoriesToUpdate;
    private List<SubCategoryDTO> subCategoriesToUpdate;
    private List<ParametersDTO> parametersToUpdate;
    private List<SelectiveParameterValueDTO> selectiveParametersToUpdate;
    private List<Long> mainCategoryIdsToDelete;
    private List<Long> subCategoryIdsToDelete;
    private List<Long> parameterIdsToDelete;
    private List<Long> selectiveParameterIdsToDelete;
}
